shell/browser/ui/views/electron_constrained_window_views_client.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/ui/views/electron_constrained_window_views_client.h"
+
+#include "base/macros.h"
+#include "components/web_modal/web_contents_modal_dialog_host.h"
+#include "shell/browser/native_window.h"
+#include "shell/browser/window_list.h"
+#include "shell/common/platform_util.h"
+
+namespace {
+
+class ElectronConstrainedWindowViewsClient
+    : public constrained_window::ConstrainedWindowViewsClient {
+ public:
+  ElectronConstrainedWindowViewsClient() {}
+  ~ElectronConstrainedWindowViewsClient() override {}
+
+ private:
+  // ConstrainedWindowViewsClient:
+  web_modal::ModalDialogHost* GetModalDialogHost(
+      gfx::NativeWindow parent) override {
+    // Get the browser dialog management and hosting components from |parent|.
+    auto windows = electron::WindowList::GetWindows();
+    for (auto* window : windows) {
+      if (window->GetNativeWindow() == parent)
+        return window->GetWebContentsModalDialogHost();
+    }
+
+    return nullptr;
+  }
+  gfx::NativeView GetDialogHostView(gfx::NativeWindow parent) override {
+    return platform_util::GetViewForWindow(parent);
+  }
+
+  DISALLOW_COPY_AND_ASSIGN(ElectronConstrainedWindowViewsClient);
```

Comment:
The Electron codebase no longer uses `DISALLOW_COPY_AND_ASSIGN` as of #31633. Could you update this PR to change these usages out?

Reply:
Yep I'll swap em out.

Reply:
Updated in 315f996

Reply:
Could you also remove the includes of "base/macros.h"? Those were only needed for `DISALLOW_COPY_AND_ASSIGN`.

Reply:
Yep!